### ISSUE TITLE
OPT/RF to handle/provide multiple files into dpkg-query -S call

### ISSUE THREAD

**Issue by yarikoptic** [ReproNim/niceman#55]
Apparently it could handle being given multiple files at once.
```shell
$> dpkg-query -S /asdkljhf /usr/bin/* /alsdkjflasf 2>&1 | head     
dpkg-query: no path found matching pattern /asdkljhf
coreutils: /usr/bin/[
0install-core: /usr/bin/0alias
0install-core: /usr/bin/0desktop
0install-core: /usr/bin/0install
0install-core: /usr/bin/0launch
0install-core: /usr/bin/0store
0install-core: /usr/bin/0store-secure-add
python: /usr/bin/2to3
python2.7: /usr/bin/2to3-2.7
```
I am not sure if we would be hitting the cmdline invocation length limit (iirc 64kB?) by passing all the files at once -- but we could discover that when we actually hit it and batch execution e.g. into batches of 1k of filenames.

Note that "no path found" are spitted out into stderr.  That might also cause a problem if there is many of them and IIRC this version of the Runner might be succeptable to getting stuck if both stdout and stderr are piped... but we could handle/workaround whenever we hit that situation.

**Comment by rbuccigrossi** [comment]
We can find the maximum command-line argument list with:

os.sysconf('SC_ARG_MAX')

Which on my machine is 2097152 = 2MB (wow.)

**Comment by rbuccigrossi** [comment]
Also, to avoid dealing with stderr, if we keep track of the files we pass in, if the file wasn't associated in stdout with a project, then I believe we can assume it wasn't found.

**Comment by yarikoptic** [comment]
sounds sensible although paranoid me usually prefers to assume less and handle the fallouts ;)  I guess if someone checked dpkg-query source to see what other messages it might spit out -- it could provide a better idea.  e.g. -- if it just pukes some other error before completing processing of other files -- it is not that they could not be assigned by it to the packages, so we must be careful to not treat all errors alike

**Comment by yarikoptic** [comment]
was fixed by #58
